File: Grooming_Reservation_System_Frontend/src/app/services/appointmentservices/appointment.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Appointment } from 'src/app/dao/appointment';
import { UserDataService } from '../userservices/userdataservice.service';
import { Address } from 'src/app/dao/address';

@Injectable({
  providedIn: 'root'
})
export class AppointmentService {
  

  url = this.userDataService.url;

  constructor(private http: HttpClient, private userDataService: UserDataService) { }

  getAllAppointments(){
    return this.http.get<Appointment[]>(`${this.url}/getAllAppointments`);
  }
  getAllAppointmentsBySalonId(salonid: any){
    return this.http.get<Appointment[]>(`${this.url}/getAllAppointmentsBySalonId/${salonid}`);
  }

  getAppointmentsByUserId(userid: number) {
    return this.http.get<Appointment[]>(`${this.url}/getAppointmentByUserId/${userid}`);
  }

  saveAppointment(appointment:Appointment,userid:number,salonid:number,serviceidstr:string,stylistid:number,addressid:number){
    return this.http.post<Appointment>(`${this.url}/saveAppointment/${userid}/${salonid}/${stylistid}/${addressid}/${serviceidstr}`,appointment);
  }

  
  getAppointmentByAppointmentId(appointmentId:number){
    return this.http.get<Appointment>(`${this.url}/getAppointmentByAppointmentId/${appointmentId}`);
  }
 
  checkStylistAvailability(appointmentdate:string,stylistid:number){
    return this.http.get<Appointment[]>(`${this.url}/checkStylistAvailability/${appointmentdate}/${stylistid}`);
  }

  updateBooking(appointmentId:number,appointment:Appointment){
    console.log("inside update booking");
    
    return this.http.put<Appointment>(`${this.url}/updateBookingByAppointmentId/${appointmentId}`,appointment); 
  }

  getAllBookedAppointments(userid:number){
    return this.http.get<Appointment[]>(`${this.url}/getAllBookedAppointments/${userid}`);
  }

  getAddressbyUserid(userid: number) {
    return this.http.get<Address[]>(`${this.url}/getAddressbyUserid/${userid}`);
  }

  updateappointmentstatus(appointmentId: number,appointment:Appointment) {
   return this.http.put<Appointment>(`${this.url}/updateCancelByAppointmentId/${appointmentId}`,appointment);
  }
  
}

File: Grooming_Reservation_System_Backend/src/main/java/com/edu/grooming/service/AppointmentService.java
package com.edu.grooming.service;

import java.time.LocalDate;
import java.util.List;
import java.util.Set;

import com.edu.grooming.dao.Appointment;
import com.edu.grooming.error.NotFoundException;


public interface AppointmentService {

	Appointment saveAppointment(Appointment appointment,String servicesid);

	List<Appointment> getAllAppointments();

	Appointment updateAppointmentUser(Integer userid, Integer appointmentId);

	Appointment updateAppointmentSalon(Integer salonid, Integer appointmentId);

	Appointment updateAppointmentStylist(Integer stylistid, Integer appointmentId);

	Appointment addServiceAppointment(Integer serviceid, Integer appointmentId) throws NotFoundException;

	Appointment getAppointmentByAppointmentId(Integer appointmentId);

	List<Appointment> getAppointmentByUserId(Integer userid);

	List<Appointment> deleteAppointmentByAppointmentId(Integer appointmentId) throws NotFoundException;

	List<Appointment> getAllAppointmentsBySalonId(Integer salonid);

	List<Appointment> checkStylistAvailability(String appointmentdate, Integer stylistid);

	Appointment updateBooking(Integer appointmentId, Appointment appointment);

	List<Appointment> getAllBookedAppointments(Integer userid);

	

}
